vscode/src/client/analyzerClient.ts

```diff
@@ -253,10 +253,16 @@ export class AnalyzerClient {
             )}`,
           );
 
-          const response: any = await this.rpcConnection!.sendRequest(
-            "analysis_engine.Analyze",
-            requestParams,
-          );
+          const cancellationPromise = new Promise((_, reject) => {
+            token.onCancellationRequested(() => {
+              reject(new Error("Operation canceled by the user."));
```

Comment:
1. rejecting is equivalent to throwing an exception -  I cannot test it but the `catch` block should get called. Consider resolving and exiting in a controlled way.
2. in the vscode code base  `token.isCancellationRequested` check is used sometimes to support the callback (I guess just-in-case the action would be cancelled before the callback was added)

Reply:
Good catch. Addressed both.